### PR TITLE
add terraform wordpress sample with kube and mysql

### PR DESCRIPTION
**Pull request by ksotter**

### CODE REVIEW

**Comment by raclepoulpe** [comment]
Hello Kilian
This tutorial need : 
- clarifying on the way to apply the different layers (describe how to apply the whole content)
- clarifying the kubernetes variables
- add region parameter on each resource
- removing personal datas from paths

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+variable service_name {
+  description = "Public cloud project service name"
+  type        = string
+  default     = ""
+}
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:5]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+variable ovh {
+  description = "OVHcloud provider parameters"
+  type    = map
+  default = {
+    endpoint           = ""
+    application_key    = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:12]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    application_secret = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:13]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    consumer_key       = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:14]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+* variables.tf   : Contains the variables we need to declare
+* secrets.tfvars : Contains the value of the variables
+* output.tf      : Contains the output we want
+* backend.tf     : Create the backend for the layering
+
+### Variables
+
+|Variable name        |Variable parameter   |Parameter description|
+|---------------------|---------------------|---------------------|
+|ovh|endpoint|API endpoint to use|
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:41]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+|ovh|application_key|API Application Key|
+|ovh|application_secret|API application secret|
+|ovh|consumer_key|API consumer key|
+|kubernetes|project_id|Public cloud project service name|
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:45]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+|kubernetes|region|Region of the Kubernetes cluster|
+|database|project_id|Public cloud project service name|
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:47]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+### Create the terraform variables file
+```console
+read -s APPLICATION_KEY
+read -s APPLICATION_SECRET
+read -s CONSUMER_KEY
+read CLOUD_PROJECT_ID
+
+cat << EOF > secrets.tfvars
+ovh = {
+    endpoint           = "ovh-eu"
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:69]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+### Create the terraform variables file
+```console
+read -s APPLICATION_KEY
+read -s APPLICATION_SECRET
+read -s CONSUMER_KEY
+read CLOUD_PROJECT_ID
+
+cat << EOF > secrets.tfvars
+ovh = {
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:68]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/main.tf [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  content  = ovh_cloud_project_kube.wordpress_kube_cluster.kubeconfig
+  filename = "kubeconfig_file"
+}
+
+# Retrieve info from nodes in the pool
+data "ovh_cloud_project_kube_nodepool_nodes" "nodes" {
+  depends_on = [
+    ovh_cloud_project_kube_nodepool.wordpress_node_pool
+  ]
+  service_name  = ovh_cloud_project_kube.wordpress_kube_cluster.service_name
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/main.tf:34]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/provider.tf [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  required_providers {
+    ovh = {
+      source  = "ovh/ovh"
+      version = "0.25.0"
+    }
+  }
+}
+
+provider "ovh" {
+  endpoint           = var.ovh.endpoint
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/provider.tf:11]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 12]
```diff
@@ -0,0 +1,25 @@
+variable service_name {
+  description = "Public cloud project service name"
+  type        = string
+  default     = ""
+}
+
+variable ovh {
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:7]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+  description = "OVHcloud provider parameters"
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:8]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+  type    = map
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:9]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+  default = {
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:10]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+    endpoint           = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:11]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+    application_key    = ""
+    application_secret = ""
+    consumer_key       = ""
+  }
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:15]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+}
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:16]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer1/variables.tf [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+
+variable kubernetes {
+  description = "Kubernetes cluster definition"
+  type    = map
+  default ={
+    project_id = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer1/variables.tf:22]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/main.tf [hunk 20]
```diff
@@ -0,0 +1,94 @@
+resource "local_file" "kubeconfig_file" {
+  content  = data.terraform_remote_state.layer1.outputs.kubeconfig_file
+  filename = "kubeconfig_file"
+}
+
+#Create the DB service
+resource "ovh_cloud_project_database" "database_service" {
+  service_name = var.database.project_id
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/main.tf:8]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/main.tf [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+  }
+  flavor       = var.database.flavor
+}
+
+# Create the managed mySQL DB
+resource "ovh_cloud_project_database_database" "wordpress_db" {
+  depends_on = [
+    ovh_cloud_project_database.database_service
+  ]
+  service_name  = var.database.project_id
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/main.tf:24]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/main.tf [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+  cluster_id    = ovh_cloud_project_database.database_service.id
+  name          = "wordpress_db"
+}
+
+# Create OVH user for the DB
+resource "ovh_cloud_project_database_user" "wordpress_db_user" {
+  depends_on = [
+    ovh_cloud_project_database_database.wordpress_db
+  ]
+  service_name = ovh_cloud_project_database.database_service.service_name
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/main.tf:35]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/main.tf [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+  id = each.value.instance_id
+}
+
+# DB IP restrictions with the IPs of the nodes
+resource "ovh_cloud_project_database_ip_restriction" "nodes_iprestriction" {
+  depends_on = [
+    ovh_cloud_project_database_database.wordpress_db
+  ]
+  for_each = data.openstack_compute_instance_v2.instances
+  service_name = ovh_cloud_project_database.database_service.service_name
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/main.tf:55]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/provider.tf [hunk 24]
```diff
@@ -0,0 +1,38 @@
+terraform {
+  required_providers {
+    ovh = {
+      source  = "ovh/ovh"
+      version = "0.25.0"
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/provider.tf:5]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/provider.tf [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    openstack = {
+      source  = "terraform-provider-openstack/openstack"
+      version = "~> 1.48.0"
+    }
+  }
+}
+
+provider "ovh" {
+  endpoint           = var.ovh.endpoint
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/provider.tf:19]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/provider.tf [hunk 26]
```diff
@@ -0,0 +1,38 @@
...
+  application_key    = var.ovh.application_key
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/provider.tf:20]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/provider.tf [hunk 27]
```diff
@@ -0,0 +1,38 @@
...
+  application_secret = var.ovh.application_secret
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/provider.tf:21]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/provider.tf [hunk 28]
```diff
@@ -0,0 +1,38 @@
...
+  consumer_key       = var.ovh.consumer_key
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/provider.tf:22]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 29]
```diff
@@ -0,0 +1,49 @@
+variable service_name {
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:1]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 30]
```diff
@@ -0,0 +1,49 @@
...
+  description = "Public cloud project service name"
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:2]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 31]
```diff
@@ -0,0 +1,49 @@
...
+  type        = string
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:3]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 32]
```diff
@@ -0,0 +1,49 @@
...
+  default     = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:4]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 33]
```diff
@@ -0,0 +1,49 @@
...
+}
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:5]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 34]
```diff
@@ -0,0 +1,49 @@
...
+
+variable ovh {
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:7]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 35]
```diff
@@ -0,0 +1,49 @@
...
+  description = "OVHcloud provider parameters"
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:8]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 36]
```diff
@@ -0,0 +1,49 @@
...
+  type    = map
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:9]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 37]
```diff
@@ -0,0 +1,49 @@
...
+  default = {
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:10]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 38]
```diff
@@ -0,0 +1,49 @@
...
+    endpoint           = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:11]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 39]
```diff
@@ -0,0 +1,49 @@
...
+    application_key    = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:12]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 40]
```diff
@@ -0,0 +1,49 @@
...
+    application_secret = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:13]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 41]
```diff
@@ -0,0 +1,49 @@
...
+    consumer_key       = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:14]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 42]
```diff
@@ -0,0 +1,49 @@
...
+  }
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:15]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 43]
```diff
@@ -0,0 +1,49 @@
...
+}
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:16]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 44]
```diff
@@ -0,0 +1,49 @@
...
+
+variable database { 
+  description = "Wordpress website database parameters"
+  type     = map
+  default = {
+    project_id = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:22]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/layer2/variables.tf [hunk 45]
```diff
@@ -0,0 +1,49 @@
...
+    flavor     = ""
+    version    = ""
+  } 
+}
+
+variable kubernetes {
+  description = "Kubernetes cluster definition"
+  type    = map
+  default ={
+    project_id = ""
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/layer2/variables.tf:34]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 46]
```diff
@@ -0,0 +1,131 @@
...
+The purpose of this tutorial is to create a mysql database and a wordpress website and link them together in a Kubernetes cluster.
+
+We will divide the project into two layers. The first layer is used to deploy the cluster and fetch the nodes. The second layer is used to deploy the services. Each layer is executed one by one.
+
+![Infrastructure diagram](./img/diagram.png)
+
+## Requirements
+
+You need the following:
+* [terraform](https://www.terraform.io/) installed
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:12]:
```suggestion
* [terraform](https://www.terraform.io/) installed
* [curl](https://curl.se/) installed
```

**Review by scraly — CHANGES_REQUESTED** [review]
Hi,
thanks for the work :).


I reviewed the code and there are a way to simplify a lot the configuration :-).

File: terraform/wordpress-k8s-mysql/README.md [hunk 47]
```diff
@@ -0,0 +1,141 @@
...
+
+## Set the environment variables
+
+```bash
+# OVHcloud provider needed keys
+export OVH_ENDPOINT="ovh-eu"
+export OVH_APPLICATION_KEY="xxx"
+export OVH_APPLICATION_SECRET="xxx"
+export OVH_CONSUMER_KEY="xxx"
+export OVH_CLOUD_PROJECT_SERVICE="xxx"
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:27]:
```suggestion
export OVH_CLOUD_PROJECT_SERVICE="xxx"
```
```

**Review by scraly — CHANGES_REQUESTED** [review]

File: terraform/wordpress-k8s-mysql/README.md [hunk 48]
```diff
@@ -0,0 +1,133 @@
...
+|openstack|tenant_name|The Name of the Tenant to login with|
+|openstack|password|The Password to login with|
+|openstack|auth_url|The Identity authentication URL|
+|openstack|region|The region of the OpenStack cloud to use|
+
+## Build and run
+
+### Create the terraform variables file
+```console
+read -s OVH_APPLICATION_KEY
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:65]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 49]
```diff
@@ -0,0 +1,133 @@
...
+read -s OVH_APPLICATION_SECRET
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:66]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 50]
```diff
@@ -0,0 +1,133 @@
...
+read -s OVH_CONSUMER_KEY
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:67]:
```suggestion
```

File: terraform/wordpress-k8s-mysql/README.md [hunk 51]
```diff
@@ -0,0 +1,133 @@
...
+read OVH_CLOUD_PROJECT_SERVICE
```
Comment (scraly) [inline terraform/wordpress-k8s-mysql/README.md:68]:
```suggestion
```

**Review by scraly — CHANGES_REQUESTED** [review]

**Review by scraly — APPROVED** [review]

**Review by scraly — APPROVED** [review]

**Pull request merged**